### DISCUSSION TITLE
Next js skips the execution of server components for no reason

### DISCUSSION THREAD

**Discussion by rforzani** [vercel/next.js#49713] [App Router] [answered]
Hi I have a profile page and a settings page for my users. The profile page retrieves the user data in the server component. I am trying to show the latest user data in the profile page, however when users try to update any data from the settings, the data successfully gets updated but next js seems to cache the server component in some way and does not rerun it, therefore providing outdated values. 
I of course have disabled cache on all fetch requests, but what's even stranger is that the whole server component is not executing even if I just place a simple console.log() in it. 
Anybody have any ideas?

**Comment by schrodienieur** [comment, marked as answer]
I think that is because when you navigate between profile and settings page, It uses soft navigation. It won't refetch the data as in the docs said. 

https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating

I also experiencing that, then I change my navigation to a function like this. 

router.push(href)
router.refresh()

router.refresh() will refetch the data. 

But it will send 2 requests per click.


Maybe this link helps. 

https://github.com/vercel/next.js/issues/42991

**Reply by rforzani** [reply]
Wow thanks a lot for the prompt reply. By looking at next js docs, it seems like router.push() uses soft navigation by default. Since this cannot be modified (as of now), the perfect workaround is to add a router.replace() function followed by a router.refresh()
